# hand_shadowing_params.yaml
hand_shadowing_node:
  ros__parameters:
    # General parameters
    scale_factor: 2.0
    smoothing_factor: 0.3
    
    # Scaling parameters by joint index
    scaling:
      thumb_cmc_abd: 2.0      # index 0
      thumb_mcp_abd: 5.0        # index 1
      finger_mcp_abd: -2.5         # indices 2-5
      finger_mcp_flex: 1.0       # indices 6-10
      thumb_mcp_flex: -5.0              # index 11
      finger_pip: -2.0           # indices 12-15 
      thumb_cmc_flex: 5.0        # index 16
    
    # Compensation parameters by joint index
    compensation:
      thumb_cmc_abd: -25.0      # index 0
      thumb_mcp_abd: 20.0        # index 1
      index_mcp_abd: 0.0          # index 2
      middle_mcp_abd: 0.0        # index 3
      ring_mcp_abd: 0.0          # index 4
      pinky_mcp_abd: 0.0        # index 5
      finger_mcp_flex: 8.0        # indices 6-10
      thumb_mcp_flex: 0.0             # index 11
      finger_pip: 30.0            # indices 12-15
      thumb_cmc_flex: 35.0        # index 16
    
    # Clamping parameters by joint index (min, max)
    clamping:
      thumb_cmc_abd: [-25.0, 60.0]      # index 0
      thumb_mcp_abd: [-60.0, 40.0]      # index 1
      finger_mcp_abd: [-25.0, 25.0]     # indices 2-4
      pinky_mcp_abd: [-40.0, 25.0]        # index 5
      finger_mcp_flex: [-90.0, 90.0]    # indices 6-10
      thumb_mcp_flex: [-50.0, 90.0]          # index 11
      finger_pip: [-50.0, 90.0]         # indices 12-15
      thumb_cmc_flex: [-85.0, 90.0]     # index 16
      default: [-90.0, 90.0]            # default